core/modules/core/include/klepsydra/core/configuration_environment.h

```diff
@@ -118,31 +115,60 @@ class YamlEnvironment : public Environment
      */
     virtual void setPropertyBool(const std::string &key,
                                  const bool &value,
-                                 const std::string &rootNode = kpsr::DEFAULT_ROOT);
+                                 const std::string &rootNode = kpsr::DEFAULT_ROOT) override;
 
     /**
-     * \brief reload
-     * \param yamlContent
+     * @brief loadFile
+     * @param fileName
+     * @param nodeName
+     *
+     * This method is used to load additional configuration data from another file. It might be used in cases where
+     * additional data may be loaded later.
      */
-    void updateConfiguration(const std::string &yamlContent);
+    virtual void loadFile(const std::string &fileName, const std::string &nodeName) override;
 
-    void updateConfiguration(const std::string &yamlContent, const std::string &rootNode);
-    /**
-     * @brief exportEnvironment
-     * @return
-     */
-    std::string exportEnvironment();
+    void updateConfiguration(const std::string &jsonContent);
 
-    virtual void loadFile(const std::string &fileName,
-                          const std::string &nodeName = kpsr::DEFAULT_ROOT);
+    void updateConfiguration(const std::string &jsonContent, const std::string &rootNode);
 
-protected:
-    const std::string _yamlFileName;
-    YAML::Node _node;
+    std::string exportEnvironment();
 
 private:
-    YAML::Node getNode(const std::string &rootNode);
+    void printGetError(bool result, const std::string &key) const;
+    void printSetError(bool result, const std::string &key) const;
+
+    template<class Archive>
+    void serialize(Archive &archive)
+    {
```

Comment:
Maybe it would be better to use just one try-catch and use e.what() to get exactly what error took place?
```suggestion
    {
      try {
          archive(cereal::make_nvp("StringProperties", _stringProperties)
          ,cereal::make_nvp("FloatProperties", _floatProperties)
          ,cereal::make_nvp("IntProperties", _intProperties)
          ,cereal::make_nvp("BoolProperties", _boolProperties)
          );
          } catch (const std::exception &e) {
            spdlog::error("[ConfigurationEnvironment::serialize] {}", e.what());
        }
```

Reply:
This is a good idea. However, at some point we will get rid of the `try / catch` blocks for compliance with ESA and freeRTOS. I'd implement it for now as it might make our lives easier.

Reply:
The proposed change fails on `kpsr_core_tests`: reads a file without `StringProperties`, raises an error, `FloatProperties` are never read so
```ctest
2: [ RUN      ] ConfigEnvironmentLogger.LoadFileMissingStringTest
2: /home/fbarros/klepsydra/src/kpsr-core/core/tests/core_tests/src/config_environment_test.cpp:425: Failure
2: Expected equality of these values:
2:   checkFloatPropertyValueTwo
2:     Which is: 0
2:   floatPropertyValueTwo
2:     Which is: 3.14535
2: [  FAILED  ] ConfigEnvironmentLogger.LoadFileMissingStringTest (1 ms)
```